crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1030,7 +1035,28 @@ pub(crate) fn handle_rename(
     if !change.file_system_edits.is_empty() && snap.config.will_rename() {
         change.source_file_edits.clear();
     }
+
     let workspace_edit = to_proto::workspace_edit(&snap, change)?;
+
+    if let Some(lsp_types::DocumentChanges::Operations(ops)) =
+        workspace_edit.document_changes.as_ref()
+    {
+        for op in ops {
+            if let lsp_types::DocumentChangeOperation::Op(doc_change_op) = op {
+                if let Err(err) = resource_ops_supported(
+                    &snap.config,
+                    match doc_change_op {
+                        ResourceOp::Create(_) => &ResourceOperationKind::Create,
+                        ResourceOp::Rename(_) => &ResourceOperationKind::Rename,
+                        ResourceOp::Delete(_) => &ResourceOperationKind::Delete,
+                    },
```

Comment:
let's move this into a function as well `fn(&ResourceOp) -> ResourceOperationKind`